lib/expression/eval.ex

```diff
@@ -60,8 +60,17 @@ defmodule Expression.Eval do
 
   def eval!({:function, opts}, context, mod) do
     name = opts[:name] || raise "Functions need a name"
-    arguments = opts[:args] || []
-    IO.inspect(mod, label: "mod")
+    args = opts[:args] || []
+
+    arguments =
+      Enum.map(args, fn
+        {type, _args} = function when type in [:function, :lambda] ->
+          value = eval!(function, context, mod)
+          [literal: value]
```

Comment:
Can you talk me through why this special casing is needed here?

Reply:
My guess is that we're doing this to force the evaluation of all function arguments scoped to the local module first before attempting to delegate to any other module?

Reply:
what will this do with something like the `if()` function that does lazy evaluation on arguments?

Reply:
hm I think our tests already cover for that.